megamek/src/megamek/common/Entity.java

```diff
@@ -4832,8 +4832,8 @@ public int getBadCriticals(int type, int index, int loc) {
                 if (ccs.isDestroyed() || ccs.isBreached() || ccs.isMissing()) {
                     if ((type == CriticalSlot.TYPE_SYSTEM) && (ccs.getIndex() == index)) {
                         hits++;
-                    } else if ((type == CriticalSlot.TYPE_EQUIPMENT) && (m.equals(ccs.getMount()) || m.equals(ccs
-                                                                                                                      .getMount2()))) {
+                    } else if ((type == CriticalSlot.TYPE_EQUIPMENT) && 
+                            ((m != null) && m.equals(ccs.getMount()) || m.equals(ccs.getMount2()))) {
```

Comment:
```suggestion
                            ((m != null) && (m.equals(ccs.getMount()) || m.equals(ccs.getMount2())))) {
```
Just to avoid future misreads, as I did when taking a first look at this solution earlier

Reply:
That's actually a logic change, since && takes precedence over ||. Good catch.